pallets/schemas/src/migration/v4.rs

```diff
@@ -14,21 +14,23 @@ use sp_runtime::TryRuntimeError;
 use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
 
 /// get known schema names for mainnet
-pub fn get_known_schemas<T: Config>() -> BTreeMap<SchemaId, Vec<u8>> {
+pub fn get_known_schemas<T: Config>() -> BTreeMap<SchemaId, (Vec<u8>, bool)> {
 	let genesis_block: BlockNumberFor<T> = 0u32.into();
 	let genesis = <frame_system::Pallet<T>>::block_hash(genesis_block);
 
 	if let DetectedChainType::FrequencyMainNet =
 		get_chain_type_by_genesis_hash(&genesis.encode()[..])
 	{
 		// only return what needs to change which for this case is only on mainnet
+		// (schema_id, name, clear prior versions)
 		BTreeMap::from([
-			(5, b"dsnp.update".to_vec()),
-			(6, b"dsnp.profile".to_vec()),
-			(7, b"dsnp.public-key-key-agreement".to_vec()),
-			(8, b"dsnp.public-follows".to_vec()),
-			(9, b"dsnp.private-follows".to_vec()),
-			(10, b"dsnp.private-connections".to_vec()),
+			(5, (b"dsnp.update".to_vec(), true)),
+			(19, (b"dsnp.update".to_vec(), false)),
+			(6, (b"dsnp.profile".to_vec(), true)),
+			(7, (b"dsnp.public-key-key-agreement".to_vec(), false)),
+			(8, (b"dsnp.public-follows".to_vec(), false)),
+			(9, (b"dsnp.private-follows".to_vec(), false)),
+			(10, (b"dsnp.private-connections".to_vec(), false)),
```

Comment:
we do have other schemas with more than 1 version that we can also clean up please take a look below

[
  {
    schemaName: dsnp.user-attribute-set
    schemaVersion: 1
    schemaId: 11
  }
  {
    schemaName: dsnp.user-attribute-set
    schemaVersion: 2
    schemaId: 20
  }
  {
    schemaName: dsnp.profile-resources
    schemaVersion: 1
    schemaId: 15
  }
  {
    schemaName: dsnp.reaction
    schemaVersion: 1
    schemaId: 4
  }
  {
    schemaName: dsnp.update
    schemaVersion: 1
    schemaId: 6
  }
  {
    schemaName: dsnp.update
    schemaVersion: 2
    schemaId: 19
  }
  {
    schemaName: dsnp.ext-content-attribute-set
    schemaVersion: 1
    schemaId: 13
  }
  {
    schemaName: dsnp.profile
    schemaVersion: 1
    schemaId: 5
  }
  {
    schemaName: dsnp.reply
    schemaVersion: 1
    schemaId: 3
  }
  {
    schemaName: dsnp.reply
    schemaVersion: 2
    schemaId: 18
  }
  {
    schemaName: dsnp.tombstone
    schemaVersion: 1
    schemaId: 1
  }
  {
    schemaName: dsnp.tombstone
    schemaVersion: 2
    schemaId: 16
  }
  {
    schemaName: dsnp.public-key-assertion-method
    schemaVersion: 1
    schemaId: 14
  }
  {
    schemaName: dsnp.broadcast
    schemaVersion: 1
    schemaId: 2
  }
  {
    schemaName: dsnp.broadcast
    schemaVersion: 2
    schemaId: 17
  }
  {
    schemaName: dsnp.dsnp-content-attribute-set
    schemaVersion: 1
    schemaId: 12
  }
]

Reply:
None of those will trigger as they are not in the list of names/schema ids

Reply:
My question was maybe it would worth considering this opportunity and clean those up also. but again it might be outside the scope of this